syntax = "proto3";
package style_bert_vits_server;

message SetTextRequest {
  string text = 1;
}

message SetTextReply {
  bool success =1;
}

message SetParamRequest {
  optional string model_name = 1;
  optional int32 model_id = 2;
  optional float length = 3;
  optional string style = 4;
  optional string style_weight = 5;
}

message SetParamReply {
  bool success =1;
}

message InterruptVoiceRequest {}

message InterruptVoiceReply {
  bool success =1;
}

message SetVoicePlayFlgRequest {
  bool flg = 1;
}

message SetVoicePlayFlgReply {
  bool success =1;
}

service StyleBertVitsServerService {
    rpc SetText(SetTextRequest)
        returns (SetTextReply);
    rpc SetParam(SetParamRequest)
        returns (SetParamReply);
    rpc InterruptVoice(InterruptVoiceRequest)
        returns (InterruptVoiceReply);
    rpc SetVoicePlayFlg(SetVoicePlayFlgRequest)
        returns (SetVoicePlayFlgReply);
}
